projects/event-lambdas/src/lib/util.ts

```diff
@@ -112,17 +113,53 @@ export const convertNDJSONToEvents = (json: string) => {
   return events;
 };
 
-export const putEventsToKinesisStream = (events: IUserTelemetryEvent[]) => {
-  const Records = events.map((event) => ({
-    Data: JSON.stringify(event),
-    // Ordering doesn't matter
-    PartitionKey: uuidv4(),
-  }));
-  const params = {
-    Records,
-    StreamName: telemetryStreamName,
-  };
-  return kinesis.putRecords(params).promise();
+const oneMegabyteInBytes = 1024 * 1024;
+export const putEventsToKinesisStream = async (events: IUserTelemetryEvent[]) => {
+  const chunkedRecords: PutRecordsRequestEntryList[] = events.reduce((acc, event) => {
+    const record = {
+      Data: JSON.stringify({
+        "@timestamp": event.eventTime,
+        ...event,
+        tags: undefined, // 'tags' is a reserved field name since logstash v8...
+        _tags: event.tags, // ... so now we re-write to _tags
+      }),
+      // Ordering doesn't matter
+      PartitionKey: uuidv4(),
+    };
+    const currentChunk = acc[acc.length - 1];
+    if(!currentChunk){
+      return [[record]]
+    }
+    else if(currentChunk.length < 500 && Buffer.byteLength(JSON.stringify(currentChunk) + JSON.stringify(record)) < oneMegabyteInBytes){
+      return [...acc.slice(0, -1), [...currentChunk, record]];
+    }
+    else {
+      return [...acc, [record]];
+    }
+  }, [] as PutRecordsRequestEntryList[]);
+
+  for (const Records of chunkedRecords) {
+    const chunkStartTime = new Date().getTime();
+    const putRecordsResult = await kinesis.putRecords({
+      Records,
+      StreamName: telemetryStreamName,
+    }).promise();
+
+    const recordsWithErrors = putRecordsResult.Records.filter(_ => !!_.ErrorCode);
+    if(recordsWithErrors.length > 0) {
+      console.error("Failed to write some records to Kinesis", recordsWithErrors)
```

Comment:
Mentioned out of band: very happy to see these failures recorded! At the point at which we were implementing this I definitely would not have been aware `PutRecords` could fail in this way.